reporters/dogstatsd.go

```diff
@@ -84,3 +102,23 @@ func NewDogStatsD(config *viper.Viper, logger *logrus.Logger) (*DogStatsD, error
 func NewDogStatsDFromClient(c dogstatsd.Client, r string) *DogStatsD {
 	return &DogStatsD{client: c, region: r}
 }
+
+func (d *DogStatsD) scheduleRestart(timeout time.Duration) {
+	if timeout.Nanoseconds() > 0 {
+		time.AfterFunc(timeout, func() {
+			if err := d.statsdClient.Close(); err != nil {
+				d.logger.Errorf("DogStatsD: failed to close statsd connection during restart: %s", err.Error())
+			}
+			
+			c, err := dogstatsd.New(d.host, d.prefix)
+			if err == nil {
+				d.statsdClient = c.Client.(*statsd.Client)
+				d.client = c
+			} else {
+				d.logger.Errorf("DogStatsD: failed to recreate dogstatsd client during restart. %s", err.Error())
+			}
+
+			d.scheduleRestart(timeout)
```

Comment:
I'm pretty sure this will build up in the heap and result in out-of-memory. A better solution is to have loop with a time.Ticker executing this. WDYT?

Reply:
Thanks for pointing out this! I'm not much familiar with the Golang time package. I'll change to `time.Ticker`

Reply:
Changed to use time.Ticker!